lib/galaxy/job_metrics/__init__.py

```diff
@@ -68,10 +69,17 @@ class RawMetric(NamedTuple):
 class JobMetrics:
     """Load and store a collection of :class:`JobInstrumenter` objects."""
 
-    def __init__(self, conf_file=None, **kwargs):
+    def __init__(self, conf_file=None, conf_dict=None, **kwargs):
         """Load :class:`JobInstrumenter` objects from specified configuration file."""
         self.plugin_classes = self.__plugins_dict()
-        self.default_job_instrumenter = JobInstrumenter.from_file(self.plugin_classes, conf_file, **kwargs)
+        if conf_file and os.path.exists(conf_file):
+            self.default_job_instrumenter = JobInstrumenter.from_file(self.plugin_classes, conf_file, **kwargs)
+        elif conf_dict or conf_dict is None:
+            if conf_dict is None:
+                conf_dict = DEFAULT_CONFIG
+            self.default_job_instrumenter = JobInstrumenter.from_dict(self.plugin_classes, conf_dict, **kwargs)
+        else:
```

Comment:
Is this reachable?  Doesn't the previous clause cover all possibilities? (conf_file is either None or not None, if we don't count the possibility of `false`, the empty string and zero?)

Reply:
It covers any non-`None` falsey values so you can set `job_metrics: false` or `job_metrics: []` and disable the default behavior of getting the `core` plugin if nothing is configured.

Reply:
Got it! Would you be terribly opposed to adding an inline comment that clarifies this? (for folks like myself :)

Reply:
Done, plus I added it to the docs for good measure.

Reply:
Awesome, thanks!